ext-apiserver/cloud/google/machineactuator.go

```diff
@@ -329,7 +329,13 @@ func (gce *GCEClient) Delete(machine *clusterv1.Machine) error {
 			"error deleting GCE instance: %v", err))
 	}
 
-	return nil
+	if gce.machineClient != nil {
+		// Remove the finalizer
+		machine.ObjectMeta.Finalizers = util.Filter(machine.ObjectMeta.Finalizers, clusterv1.MachineFinalizer)
+		_, err = gce.machineClient.Update(machine)
```

Comment:
does the update call do retries if it fails?

Reply:
I don't think so. But as long as finalizer is not being removed, reconciliation will keep going, it's effectively being retried.

Reply:
In the mean time, reconciliation loop built on top of apiserver builder had built-in retries already.